godel_robots/abb/abb_file_suite/src/ftp_upload.cpp

```diff
@@ -0,0 +1,139 @@
+#include "ftp_upload.h"
+
+#include <stdlib.h>
+#include <stdio.h>
+
+#include <curl/curl.h>
+
+/* parse headers for Content-Length */
+static size_t getcontentlengthfunc(void *ptr, size_t size, size_t nmemb, void *stream)
+{
+  int r;
+  long len = 0;
+
+  /* _snscanf() is Win32 specific */
+  r = sscanf(static_cast<const char*>(ptr), "Content-Length: %ld\n", &len);
+
+  if (r) /* Microsoft: we don't read the specs */
+    *((long *) stream) = len;
+
+  return size * nmemb;
+}
+
+/* discard downloaded data */
+static size_t discardfunc(void *ptr, size_t size, size_t nmemb, void *stream)
+{
+  return size * nmemb;
+}
+
+/* read data to upload */
+static size_t readfunc(void *ptr, size_t size, size_t nmemb, void *stream)
+{
+  FILE *f = static_cast<FILE*>(stream);
+  size_t n;
+
+  if (ferror(f))
+    return CURL_READFUNC_ABORT;
+
+  n = fread(ptr, size, nmemb, f) * size;
+
+  return n;
+}
+
+
+static int upload(CURL *curlhandle, const char * remotepath, const char * localpath,
+           long timeout, long tries)
+{
+  FILE *f;
+  long uploaded_len = 0;
+  CURLcode r = CURLE_GOT_NOTHING;
+  int c;
```

Comment:
At the expense of sounding repetitive, using short but descriptive words in naming variables will help with code readability.